lib/org/chromium/webidl/DAOOperation.js

```diff
@@ -0,0 +1,59 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'DAOOperation',
+
+  properties: [
+    {
+      class: 'String',
+      documentation: 'DAO method name associated with operation.',
+      name: 'methodName',
+    },
+    {
+      documentation: 'Arguments object associated with operation.',
+      name: 'args',
+    },
+    {
+      name: 'promise_',
+      factory: function() {
+        var self = this;
+        var resolve;
+        var reject;
+        var promise = new Promise(function(res, rej) {
```

Comment:
We should make this pattern into a reusable 'Future' object. (Not now, just more of a note to myself.)